### PR TITLE
v0.20 - Probably Australian

### PR DESCRIPTION
**Pull request by shiftkey**
- [x] release notes
- [x] version bump
- [x] integration tests all pass
- [x] :thumbsup: from @Haacked or @ryangribble 
- [x] tag and publish to NuGet
- [ ] :shipit:

### CODE REVIEW

File: Octokit.Tests.Integration/Clients/RepositoryHooksClientTests.cs [hunk 1]
```diff
@@ -276,7 +276,7 @@ public TheDeleteMethod(RepositoriesHooksFixture fixture)
                 await github.Repository.Hooks.Delete(_fixture.RepositoryOwner, _fixture.RepositoryName, _fixture.ExpectedHook.Id);
                 var hooks = await github.Repository.Hooks.GetAll(_fixture.RepositoryOwner, _fixture.RepositoryName);
 
-                Assert.Empty(hooks);
+                Assert.Equal(4, hooks.Count);
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/RepositoryHooksClientTests.cs]: Each test creates the same five hooks, so we need to verify the count of the list to be `N-1`

File: Octokit.Tests.Integration/Clients/RepositoryHooksClientTests.cs [hunk 2]
```diff
@@ -196,7 +196,7 @@ public TheEditMethod(RepositoriesHooksFixture fixture)
                 var actualHook = await github.Repository.Hooks.Edit(_fixture.RepositoryOwner, _fixture.RepositoryName, _fixture.ExpectedHook.Id, editRepositoryHook);
 
                 var expectedConfig = new Dictionary<string, string> { { "content_type", "json" }, { "url", "http://test.com/example" } };
-                Assert.Equal(new[] { "commit_comment", "pull_request" }.ToList(), actualHook.Events.ToList());
+                Assert.Equal(new[] { "deployment", "pull_request" }.ToList(), actualHook.Events.ToList());
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/RepositoryHooksClientTests.cs]: This property is unrelated to the test itself, so cleaning it up...

File: Octokit.Tests.Integration/Clients/RepositoryContentsClientTests.cs [hunk 3]
```diff
@@ -192,7 +192,7 @@ public class TheGetContentsMethod
             var archive = await github
                 .Repository
                 .Content
-                .GetArchive("alfhenrik", "ScriptCs.OctoKit", ArchiveFormat.Tarball, "dev");
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/RepositoryContentsClientTests.cs]: This branch was removed on the repository :cry:

File: Octokit.Tests.Integration/Clients/RepositoryHooksClientTests.cs [hunk 4]
```diff
@@ -214,7 +214,7 @@ public TheEditMethod(RepositoriesHooksFixture fixture)
                 var actualHook = await github.Repository.Hooks.Edit(_fixture.RepositoryOwner, _fixture.RepositoryName, _fixture.ExpectedHook.Id, editRepositoryHook);
 
                 var expectedConfig = new Dictionary<string, string> { { "project", "GEZDGORQFY2TCNZRGY2TSMBVGUYDK" } };
-                Assert.Equal(new[] { "commit_comment", "pull_request" }.ToList(), actualHook.Events.ToList());
+                Assert.Equal(new[] { "deployment", "pull_request" }.ToList(), actualHook.Events.ToList());
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/RepositoryHooksClientTests.cs]: ⬆️

File: Octokit.Tests.Integration/Reactive/ObservableEventsClientTests.cs [hunk 5]
```diff
@@ -169,10 +169,18 @@ public TheGetAllIssuesForRepositoryMethod()
             {
                 _eventsClient = new ObservableEventsClient(Helper.GetAuthenticatedClient());
             }
+
             [IntegrationTest]
             public async Task ReturnsRepositoryEvents()
             {
-                var repositoryEvents = await _eventsClient.GetAllIssuesForRepository(owner, name).ToList();
+                var options = new ApiOptions
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Reactive/ObservableEventsClientTests.cs]: Not really needed, but I found this while looking into #1374 / #1251

**Comment by shiftkey** [comment]
Pending any feedback on the words from @Haacked or @ryangribble this is good to go.

Aiming to do it early Wedneday AEST so I have the day to monitor it in case something unexpected occurs...

File: Octokit.Tests.Integration/Clients/RepositoryContentsClientTests.cs (hunk 3 above)
Comment (alfhenrik) [inline Octokit.Tests.Integration/Clients/RepositoryContentsClientTests.cs]: 💩 sorry! I can add it back in, or I can update the test to use the `master` branch instead (as that **should never** be removed 😇

Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/RepositoryContentsClientTests.cs]: @alfhenrik all good, once I realised what was happening it was an easy fix 😀

**Comment by ryangribble** [comment]
> Pending any feedback on the words 

"Add Migrations API for Enterprise instances".  
This is a bit ambiguous as the API does work on github.com and GHE.  Also this is a preview API so may as well mention that, since it's mentioned on various other items.
Suggest rewording to "Add Migrations Preview API" or "Add Migrations Preview API for migrating repositories to GitHub Enterprise"

The details on #1139 has via ryangribble but should be via @M-Zuber 

Looking at #1139 again I think we also may have made a breaking change (renaming GetAll() to GetALlForCurrent without obsoleting the old metho?)

I also recall us having other changes where we `[Obsoleted]` stuff but I can't track them down at the moment!

File: ReleaseNotes.md [hunk 6]
```diff
@@ -1,3 +1,60 @@
+### New in 0.20.0 (released 2016/06/15)
+
+**Features**
+
+Pagination support has been added, letting you control how much data you want
```
Comment (Haacked) [inline ReleaseNotes.md]: "has been added" is passive voice and kind of redundant. How about

> Pagination  support lets  you control how much ...

Comment (Haacked) [inline ReleaseNotes.md]: Ah, after looking down, I see that "Pagination Support" is being called out in a special manner and isn't in the bulleted list. Maybe we should call out why. For example:

> The big focus of this release is pagination support. Pagination support lets you control...

**Comment by Haacked** [comment]
Just a couple minor nits  on the release notes. Otherwise, this release is exciting!

**Comment by shiftkey** [comment]
@Haacked @ryangribble thanks for the feedback, I'll incorporate that once I'm done with breakfast.

**Comment by shiftkey** [comment]
@ryangribble I'll do another pass over the PRs merged to see if I can find other notable `[Obsolete]` changes - I think #1224 is relevant here

**Comment by shiftkey** [comment]
:ship:ed

 - https://www.nuget.org/packages/Octokit/0.20.0
 - https://www.nuget.org/packages/Octokit.Reactive/0.20.0
